### PR TITLE
remove extra call to get_viz in explorev2

### PR DESCRIPTION
**Pull request by vera-liu**
Issue:
 - Currently we make an ajax call to update_explore endpoint with form_data, which calls get_viz in the backend, later in render() we make ajax call to explore_json endpoint, so get_viz is called two times for each explore update

Change:
 - update json endpoint, csv endpoint and standalone endpoint with new form_data once query button is clicked, remove update_explore endpoint
 - pass back payload with query from slice.done() method in mocked slice, to update Query button in action buttons

needs-review @mistercrunch @ascott

### CODE REVIEW

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx [hunk 1]
```diff
@@ -55,7 +55,13 @@ class ChartContainer extends React.Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    this.setState({ mockSlice: this.getMockedSliceObject(nextProps) });
+    if (nextProps.chartStatus !== this.props.chartStatus) {
```
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: should there be a ` && nextProps.chartStatus !== 'running'` here?

File: superset/assets/javascripts/explorev2/components/ChartContainer.jsx [hunk 2]
```diff
@@ -166,16 +174,19 @@ class ChartContainer extends React.Component {
...
-      />
+      <div>
+        {loading &&
+          <img alt="loading" width="25" src="/static/assets/images/loading.gif" />
+        }
+        <div
+          id={this.props.containerId}
+          ref={(ref) => { this.chartContainerRef = ref; }}
+          className={this.props.viz_type}
+          style={{ overflowX: 'scroll' }}
```
Comment (mistercrunch) [inline superset/assets/javascripts/explorev2/components/ChartContainer.jsx]: [optional] this causes a bar to always show, should probably be `auto`

**Review by mistercrunch — APPROVED** [review]
Well done! This is exactly what we spoke about.

I'll happily solve the merge conflicts on my side after you merge this since this is much more important.

**Comment by ascott** [comment]
LGTM, thanks for fixing this!

**Comment by mistercrunch** [comment]
Oh I think you can now also remove the `update_chart` endpoint on the backend

**Pull request merged**